packages/playwright/src/plugins/plugin.ts

```diff
@@ -0,0 +1,240 @@
+import { readdirSync } from 'fs';
+import { basename, dirname, join, relative } from 'path';
+
+import {
+  CreateNodes,
+  CreateNodesContext,
+  joinPathFragments,
+  TargetConfiguration,
+} from '@nx/devkit';
+import { getNamedInputs } from '@nx/devkit/src/utils/get-named-inputs';
+
+import type { PlaywrightTestConfig } from '@playwright/test';
+import { getFilesInDirectoryUsingContext } from 'nx/src/utils/workspace-context';
+import minimatch = require('minimatch');
+import { loadPlaywrightConfig } from '../utils/load-config-file';
+
+export interface PlaywrightPluginOptions {
+  targetName?: string;
+  ciTargetName?: string;
+}
+
+interface NormalizedOptions {
+  targetName: string;
+  ciTargetName?: string;
+}
+
```

Comment:
Missing caching for targets, which will add 200 ms each time the graph is calculated without daemon.

See https://github.com/nrwl/nx/blob/master/packages/next/src/plugins/plugin.ts#L28-L52